### DISCUSSION TITLE
How to get Milkdown to be a fully functioning Markdown Editor

### DISCUSSION THREAD

**Discussion by PizzaConsole** [Milkdown/milkdown#1380] [General]
So its been a few months off and on trying to use Milkdown and honestly I have not spent too much time with it. But I finally just grasped the concept today. Its a Framework to build your own editor exactly as you see fit. (seems obvious I know)

I really really just did not understand fully, you have to build EVERYTHING yourself. All the menus all the buttons, even the rendering of the markdown. You have to connect all the functionality too. I was really hoping that it would abstract some of that for me. 

This is just my feedback on it. I know that it is super powerful, but the lack of complete complex example makes this very difficult for me to spend the time to rig together. The playground is nice but it only in React, where I use Vue3 and the "Recipes" are the typical programming cliché type "hello world" examples that do not really show you how to use any of the framework in a practical sense at all.

By no means am I trying to put down the work that has been done here, I am just trying to share another perspective. I will continue on this journey and maybe I can be the one to help bridge the gap for others.

**Comment by Saul-Mirone** [comment]
Thanks for your sharing. I'm really happy to see someone like you can give such a feedback.

In fact in the milkdown v6 version. Everything can work out of box but it's hard to customize. Which makes it hard to integrate into different design systems. So v7 is turned to headless pattern. Which will provide many fundamental pieces. Make it possible to let users to decide almost everything. However, it's really hard to get started if there's nothing works out of box. So the plan is:

1. Provide fundamental plugins (which has been done)
2. Provide components built on top of plugins (WIP)
    In fact I've already made some progress here.
    You can see the code here: https://github.com/Milkdown/milkdown/tree/main/packages/components
    And storybook: https://milkdown-storybook.vercel.app/?path=/story/components-code-block--javascript
3. The final plan is to provide some editors out of box, and also as a good example to show how to use milkdown (WIP)
    The current progress is crepe editor.
    You can see the code here: https://github.com/Milkdown/milkdown/tree/main/packages/crepe
    And storybook: https://milkdown-storybook.vercel.app/?path=/story/crepe-classic--width-default-value

But it's not easy to maintain and make progressive in a fully open source project. After all I don't have any reliable income from this project so I have to put my family and my job before this project. If the project can someday break even, things will be much easier. But it's still not possible now.

**Reply by PizzaConsole** [reply]
Thank you for the response! I really appreciate the information and it is extremely insightful. Again this just leans into my lack of experience with the project. But I hope I can continue progressing with using it.

> But it's not easy to maintain and make progressive in a fully open source project. After all I don't have any reliable income from this project so I have to put my family and my job before this project. If the project can someday break even, things will be much easier. But it's still not possible now.

I 100% agree. This is where the community (including myself) needs to step up and contribute to this amazing project!
